.github/workflows/testing.yml

```diff
@@ -0,0 +1,33 @@
+permissions:
+    contents: read
+on:
+    push:
+        branches: [main]
+    pull_request:
+# If new code is pushed to a PR branch, then cancel in progress workflows for that PR. Ensures that
+# we don't waste CI time, and returns results quicker https://github.com/jonhoo/rust-ci-conf/pull/5
+concurrency:
+    group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}
+    cancel-in-progress: true
+name: test
+jobs:
+    unit:
+        runs-on: ubuntu-latest
+        name: Unit tests
+        env:
+            # This is a fake connection string that is placed just for the compiler to stop whining
+            POSTGRES_CONNECTION_STRING: 'postgresql://username:password@host/db_name'
```

Comment:
Can we create a `.github/workflows/.env.ci` and have the CI source that file (`source <file>`), so we don't have to copy this to every CI we make and add every variable each time?  For example, if I add a new env var, I'd have to add it to testing, build, and docs.

Reply:
Just a note for future reference: When doing the source `.env.example` for the `docs` and `build` CI jobs, they both failed because they couldn't see the environment variables.